File: PhantomCode/src/main/java/org/firstinspires/ftc/teamcode/own/opmodes/teleop/MainTeleOp.java
package org.firstinspires.ftc.teamcode.own.opmodes.teleop;

import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import org.firstinspires.ftc.teamcode.own.actions.gamepadaction.CaptureGamepad;
import org.firstinspires.ftc.teamcode.own.actions.gamepadaction.LaunchGamepad;
import org.firstinspires.ftc.teamcode.own.actions.gamepadaction.WheelBaseAction;
import org.firstinspires.ftc.teamcode.own.actions.stateaction.CaptureStateSwap;
import org.firstinspires.ftc.teamcode.own.actions.stateaction.LaunchStateSwap;
import org.firstinspires.ftc.teamcode.own.actions.stateaction.RevolverStateSwap;
import org.firstinspires.ftc.teamcode.own.mechanism.CaptureMechanism;
import org.firstinspires.ftc.teamcode.own.mechanism.ImuMechanism;
import org.firstinspires.ftc.teamcode.own.mechanism.LaunchMechanism;
import org.firstinspires.ftc.teamcode.own.mechanism.StandingMechanism;
import org.firstinspires.ftc.teamcode.own.mechanism.WheelBaseMechanism;
import org.firstinspires.ftc.teamcode.own.utils.PhantomOpMode;
import org.firstinspires.ftc.teamcode.own.utils.Robot;
import org.firstinspires.ftc.teamcode.own.utils.actions.ParallelGroup;
import org.firstinspires.ftc.teamcode.own.utils.states.RotateState;

@TeleOp
public class MainTeleOp extends PhantomOpMode {
    @Override
    public void customOpModeSettings() throws InterruptedException {
        Robot.INSTANCE.addMechanism(new CaptureMechanism());

//        Robot.INSTANCE.addMechanism(new LaunchMechanism(this.hardwareMap));
//        Robot.INSTANCE.addMechanism(new StandingMechanism(this.hardwareMap));
//        Robot.INSTANCE.addMechanism(new WheelBaseMechanism(this.hardwareMap));
//        Robot.INSTANCE.addMechanism(new ImuMechanism(this.hardwareMap));
        Robot.INSTANCE.setStartAction(new ParallelGroup(
                new CaptureGamepad(),
                new RevolverStateSwap(),
                new CaptureStateSwap()
//                new LaunchGamepad(this.gamepad1, this.gamepad2),
//                new LaunchStateSwap(),
//                new WheelBaseAction(this.gamepad1,this.gamepad2)
        ));
    }
}

File: PhantomCode/src/main/java/org/firstinspires/ftc/teamcode/own/opmodes/teleop/MAIN_TELEOP.kt
package org.firstinspires.ftc.teamcode.own.opmodes.teleop
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.util.ElapsedTime
import org.firstinspires.ftc.teamcode.own.Mechanism.HorizontalSlider
import org.firstinspires.ftc.teamcode.own.Mechanism.LynxModule
import org.firstinspires.ftc.teamcode.own.Mechanism.WheelBase
import org.firstinspires.ftc.teamcode.own.Mechanism.Zaxvat

@TeleOp(name = "A_TELEOP_BASED")
class MAIN_TELEOP: LinearOpMode() {
    val bool = true
    val time = ElapsedTime()
    val lynxModule = LynxModule(this)
    val wheelBase = WheelBase(this)
    val horizontalSlider = HorizontalSlider(this)
    val zaxvat = Zaxvat(this)
    override fun runOpMode() {
        lynxModule.init_Lynx()
        wheelBase.initWheelBase(this.hardwareMap)
        horizontalSlider.init()
        zaxvat.init()
        waitForStart()
        time.reset()
        while(opModeIsActive()){
            if (bool == true){
                wheelBase.driveEasy(this.gamepad1)
            }
            else{
                wheelBase.driveFieldCentric(this.gamepad1)
            }
            zaxvat.run()
            horizontalSlider.run()
        }
    }
}